android/rctmgl/src/main/java/com/mapbox/rctmgl/components/mapview/RCTMGLMapView.java

```diff
@@ -405,11 +405,14 @@ public void selectAnnotation(RCTMGLPointAnnotation annotation) {
 
         if (id != mActiveMarkerID) {
             final MarkerView markerView = annotation.getMarker();
-            mMap.selectMarker(markerView);
+            String title = annotation.getTitle();
+            String snippet = annotation.getSnippet();
             annotation.onSelect(true);
             mActiveMarkerID = id;
-
             RCTMGLCallout calloutView = annotation.getCalloutView();
+            if (title != null || snippet !== null || (!markerView.isInfoWindowShown() && calloutView != null)) {
```

Comment:
I think instead of this `annotation.getCalloutView` should just return `null` if it does not have a title or snippet.

What do you think?

Reply:
I'm not sure. will this effect the Callout component in an annotation?
such as, if the Annotation has no title or snippet, but it has a child Callout.